apps/emqx_cluster_link/src/emqx_cluster_link_app.erl

```diff
@@ -13,14 +13,20 @@
 start(_StartType, _StartArgs) ->
     ok = mria:wait_for_tables(emqx_cluster_link_extrouter:create_tables()),
     emqx_cluster_link_config:add_handler(),
-    LinksConf = enabled_links(),
+    LinksConf = emqx_cluster_link_config:links(),
     _ =
-        case LinksConf of
+        case emqx_cluster_link_config:links() of
+            %% Even if all the links are disabled, we still want to add the hook
+            %% and register the broker, so that the feature is ready to accept
+            %% remote link requests from the configured links.
+            %% I.e, the link to `b` is disabled on the current cluster `a`,
+            %% but the link to `a` is enabled on `b`.
+            %% `b` should be able to reach `a` and establish an unidirectional link with it.
```

Comment:
Simplified as follows:
disabling/removing link on one side effectively breaks it for both sides (the local cluster stops replicating its routes and stops msg forwarding resource, which means that the remote cluster can't receive any messages from it). 
A user is however expected to eventually  disable/remove the link on both sides, so that both clusters can free up all the resources.